File: ds_&_algorithms/divide_and_conquer_algorithm/go/divide_and_conquer_algorithm.go
package main

import "fmt"

func recursive_function(n int) int {
	if n != 0 {
		return n * recursive_function(n-1)
	} else {
		return 1
	}
}

func main() {
	var number int = 4
	var result int = recursive_function(number)
	fmt.Printf("%d factorial = %d", number, result)
}

File: ds_&_algorithms/divide_and_conquer_algorithm/python/divide_and_conquer_algorithm.py
def recursive_function(n):
    if (n != 0):
        return n * recursive_function(n-1)
    else:
        return 1
    
if __name__ == "__main__":
    number = 4
    result = recursive_function(number)
    print(f'{number} & factorial = {result}')
